### PR TITLE
rustfmt 1.5.2

### PR DESCRIPTION
**Pull request by Porkepix**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  ==> rustfmt --check ./src/main.rs
  dyld[81216]: Library not loaded: '@rpath/librustc_driver-f64ac9ee9a3f0eb4.dylib'
    Referenced from: '/opt/homebrew/Cellar/rustfmt/1.5.2/bin/rustfmt'
    Reason: tried: '/usr/local/lib/librustc_driver-f64ac9ee9a3f0eb4.dylib' (no such file), '/usr/lib/librustc_driver-f64ac9ee9a3f0eb4.dylib' (no such file)
```

**Comment by SMillerDev** [comment]
Is this an issue with rustup usage?

**Comment by carlocab** [comment]
> Is this an issue with rustup usage?

Most likely, yes. We're using a pretty old nightly to build this, though. Maybe the `rust` formula works now? It may be a runtime dep.

**Comment by chenrui333** [comment]
> > Is this an issue with rustup usage?
> 
> Most likely, yes. We're using a pretty old nightly to build this, though. Maybe the `rust` formula works now? It may be a runtime dep.

yeah, it depends on a nightly build, I tried the specified nightly build, but did not quite work out for me.

**Comment by chenrui333** [comment]
the build went fine in my local, but in runtime it reference some librustc_driver

```
==> rustfmt --check ./src/main.rs
Last 15 lines from /Users/rui/Library/Logs/Homebrew/rustfmt/test.02.rustfmt:
2023-01-26 10:32:30 -0500

rustfmt
--check
./src/main.rs

dyld[52306]: Library not loaded: @rpath/librustc_driver-f64ac9ee9a3f0eb4.dylib
  Referenced from: <2CCD3F33-AAEE-3ADE-A78C-AC78D489E170> /opt/homebrew/Cellar/rustfmt/1.5.2/bin/rustfmt
  Reason: tried: '/System/Volumes/Preboot/Cryptexes/OS@rpath/librustc_driver-f64ac9ee9a3f0eb4.dylib' (no such file), '/usr/local/lib/librustc_driver-f64ac9ee9a3f0eb4.dylib' (no such file), '/usr/lib/librustc_driver-f64ac9ee9a3f0eb4.dylib' (no such file, not in dyld cache)
Error: rustfmt: failed
```

File: Formula/rustfmt.rb [hunk 1]
```diff
@@ -25,8 +25,8 @@ def install
     system "#{Formula["rustup-init"].bin}/rustup-init", "-qy", "--no-modify-path"
     ENV.prepend_path "PATH", HOMEBREW_CACHE/"cargo_cache/bin"
     # we are using nightly because rustfmt requires nightly in order to build from source
-    # pinning to nightly-2021-11-08 to avoid inconstency
-    nightly_version = "nightly-2021-11-08"
+    # pinning to nightly-2023-01-24 to avoid inconstency
+    nightly_version = "nightly-2023-01-24"
```
Comment (carlocab) [inline Formula/rustfmt.rb:29]: I wonder if we can just read this from the `rust-toolchain` file so that this doesn't have to be updated manually.

Comment (chenrui333) [reply Formula/rustfmt.rb:29]: that makes sense.

Comment (branchvincent) [reply Formula/rustfmt.rb:29]: note rustup's `cargo` already [reads the version](https://rust-lang.github.io/rustup/overrides.html) from that file, so unless we want to be explicit all we need is

```ruby
    system "#{Formula["rustup-init"].bin}/rustup-init", "-qy", "--no-modify-path"
    ENV.prepend_path "PATH", HOMEBREW_CACHE/"cargo_cache/bin"
    system "cargo", "install", *std_cargo_args
```

<details>
<summary><b>Example</b></summary>
<p>

```console
==> cargo install --locked --root /opt/homebrew/Cellar/rustfmt/1.5.2 --path .
info: syncing channel updates for 'nightly-2023-01-24-aarch64-apple-darwin'
info: latest update on 2023-01-24, rust version 1.69.0-nightly (c8e6a9e8b 2023-01-23)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'llvm-tools'
info: downloading component 'rust-docs'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustc-dev'
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: installing component 'clippy'
info: installing component 'llvm-tools'
info: installing component 'rust-docs'
info: installing component 'rust-std'
info: installing component 'rustc'
info: installing component 'rustc-dev'
info: installing component 'rustfmt'
  Installing rustfmt-nightly v1.5.2 (/private/tmp/rustfmt-20230126-48518-ucx7qv/rustfmt-1.5.2)
...
```

</p>
</details>

**Comment by chenrui333** [comment]
#137033

**Pull request closed without merging**